src/modules/formatter.rs

```diff
@@ -0,0 +1,73 @@
+use std::{io::{BufWriter, Write}, process::{Command, Stdio}};
+
+
+/// This mechanism is built to support multiple formatters.
+/// 
+/// The idea is that amber should find the one installed, verify that its compatible and use the best one possible.
+#[derive(Debug, Clone, Copy)]
+#[allow(non_camel_case_types)]
+pub enum BashFormatter {
+    /// https://github.com/mvdan/sh
+    shfmt
+}
+
+impl BashFormatter {
+    /// Get all available formatters, ordered by: best ones at the start, worser at the end
+    pub fn get_all() -> Vec<BashFormatter> {
+        vec![
+            BashFormatter::shfmt
+        ]
+    }
+
+    /// Get available formatter
+    pub fn get_available() -> Option<BashFormatter> {
+        let all = Self::get_all();
+        all.iter().find(|x| x.is_available()).map(|x| *x)
```

Comment:
I think this looks nicer

```suggestion
    pub fn get_available() -> Option<BashFormatter> {
        Self::get_all()
          .iter()
          .find(|fmt| fmt.is_available())
          .map(|fmt| *fmt)
```